tests/test_cli.py

```diff
@@ -79,7 +79,7 @@ def test_init_multiple_output_matrix(testing_workdir):
     # though - loops within outputs are contained in those top-level configs.
     matrix_dir_len = len(os.listdir(matrix_dir))
     assert matrix_dir_len == 13
-    linux_libpng16 = os.path.join(matrix_dir, "linux_libpng1.6libpq9.5.yaml")
+    linux_libpng16 = os.path.join(matrix_dir, "linux_1.6_9.5.yaml")
```

Comment:
?

Reply:
Yes, that's an unfortunate example. But is something like this really happening where we render for something like this? Typically, these look more like the other example: `win_2.7_vs2008_win32` where you can deduce the keys. We could also add a list of boring keys…but maybe that's just over engineered and the status quo is good enough.

Reply:
If we do this at all, we should shorten boring keys so that `python3.6` goes to `py3.6` etc but leave more exotic stuff fully expanded.

Reply:
Note that in the current proposal `python3.6___cpython` becomes `3.6_cpython`.

Reply:
Sure. Leaving py there is helpful.